data/delegation/store.go

```diff
@@ -0,0 +1,391 @@
+package delegation
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/Oneledger/protocol/data/balance"
+	"github.com/Oneledger/protocol/data/keys"
+	"github.com/Oneledger/protocol/serialize"
+	"github.com/Oneledger/protocol/storage"
+)
+
+type DelegationStore struct {
+	state  *storage.State
+	szlr   serialize.Serializer
+	prefix []byte
+	mux    sync.Mutex
+}
+
+type MatureBlock struct {
+	Height int64
+	Data   []*MatureData
+}
+
+type MatureData struct {
+	Address keys.Address
+	Amount  balance.Amount
+	Height  int64
+}
+
+func NewDelegationStore(prefix string, state *storage.State) *DelegationStore {
+	return &DelegationStore{
+		state:  state,
+		prefix: storage.Prefix(prefix),
+		szlr:   serialize.GetSerializer(serialize.PERSISTENT),
```

Comment:
This szlr is indented to replace the invoking of "serialize.GetSerializer(serialize.PERSISTENT)" but actually never used.